controllers/applicationvolumereplication_controller.go

```diff
@@ -417,9 +420,49 @@ func (a *AVRInstance) processPausedSubscription(
 		return IsManifestInAppliedState(pvMW)
 	}
 
+	// TODO: remove the following calls when @BenamarMk's changes go in: getVRGFromManagedCluster, isVRGReadyForFailback
+	vrg, err := a.reconciler.getVRGFromManagedCluster(subscription.Name, subscription.Namespace, newHomeCluster)
+	if err != nil {
+		return !unpause
+	}
+
+	if !a.reconciler.isVRGReadyForFailback(vrg) {
+		return !unpause
+	}
+
 	return a.cleanupAndRestore(subscription, newHomeCluster)
 }
 
+func (r *ApplicationVolumeReplicationReconciler) getVRGFromManagedCluster(
+	name string, namespace string, managedCluster string) (*rmn.VolumeReplicationGroup, error) {
+	// get VRG and verify status through ManagedClusterView
+	mcvMeta := metav1.ObjectMeta{
+		Name:      "mcv-avr-reconciler",
+		Namespace: managedCluster,
+	}
+
+	mcvViewscope := fndv2.ViewScope{
+		Resource:  "VolumeReplicationGroup",
+		Name:      name,
+		Namespace: namespace,
+	}
+
+	vrg := &rmn.VolumeReplicationGroup{}
+
+	err := r.getManagedClusterResource(mcvMeta, mcvViewscope, vrg)
+
+	return vrg, err
+}
+
+func (r *ApplicationVolumeReplicationReconciler) isVRGReadyForFailback(
```

Comment:
(nothing to do here.  Just FYI) This function is fine here to satisfy the compiler and testing, but it will not used.